Makefile

```diff
@@ -412,6 +412,7 @@ chlog-update: $(CHLOGGEN)
 genotelcontribcol: $(BUILDER)
 	./internal/buildscripts/ocb-add-replaces.sh otelcontribcol
 	$(BUILDER) --skip-compilation --config cmd/otelcontribcol/builder-config-replaced.yaml
+	bash ./internal/buildscripts/add-deprecated-factory.sh
```

Comment:
We shouldn't add any scripts here. Just list the deprecated component in cmd/otelcontribcol/builder-config.yaml

Reply:
the problem is that we said it should be subpackage and the builder uses static analysis to find NewFactory() functions. It scans the root package of each listed module, not subdirectories/subpackages. Since deprecated/ is a separate package, its NewFactory() isn’t found automatically.

Reply:
@dmitryax done